### DISCUSSION TITLE
codepen console does not always display js errors

### DISCUSSION THREAD

**Discussion by markfirmware** [alpine-collective/alpinejs-devtools#144] [Remnant]
I am posting this here since codepen is a kind of development tool and is often used to resolve alpine app coding problems.

Some, but not all, js errors from alpinejs execution are reported (with attribution to alpine) in the codepen console. Should this be reported to codepen? Does alpinejs have enough clout to prompt an improvement? Maybe report this defect along with any other requests like an option to clear the console before every run?

Thanks, Mark

**Comment by HugoDF** [comment]
The devtools don't support codepen at the moment, see https://github.com/alpine-collective/alpinejs-devtools/discussions/143#discussioncomment-273866 for alternatives.

This is a community project (that's why this repository is under alpine-collective and not the alpinejs organisation), so even if "alpinejs have enough clout to prompt an improvement", I don't think it's the place of this project to attempt to use that clout.

**Reply by markfirmware** [reply]
> This is a community project (that's why this repository is under alpine-collective and not the alpinejs organisation), so even if "alpinejs have enough clout to prompt an improvement", I don't think it's the place of this project to attempt to use that clout.

Ok. Do you think alpinejs discord is a reasonable place to discuss this?

**Reply by HugoDF** [reply]
The Discord is also community-supported, you could try to get in touch with Caleb (although I'm not sure he's interesting in pressuring codepen), maybe also get in touch with someone at codepen?

**Comment by markfirmware** [comment]
Ok. As things stand, a codepen is (one of) the official alpinejs tools (that is, a codepen is nearly universally requested for problem resolution by significant particpants in discord etc) and its console is defective. Correct me if I am wrong here. I will give this a rest and revisit in a couple of days. The codepen console (admittedly outstream) is the only defect in alpinejs that I can see right now. I want to reach a point of closure for an alpinejs diagnostics codepen add-in this week and a codepen system defect cannot be ignored.

One way to remediate it is for the diagnostic to unconditionally log something like "Not all alpinejs javascript errors will be reported here. Be sure to use the browser log to see any and all errors."

I think I will just do that for closure this week.

Thanks.

**Reply by HugoDF** [reply]
Not sure what you mean about codepen being an "official alpinejs tool", it's not affiliated with Alpine and the reason we ask for a codepen is that it's a handy way to share the code.

**Reply by markfirmware** [reply]
> Not sure what you mean about codepen being an "official alpinejs tool", it's not affiliated with Alpine and the reason we ask for a codepen is that it's a handy way to share the code.

Ok. I retract the word official. I appreciate the guidance.

**Comment by KevinBatdorf** [comment]
> a codepen is nearly universally requested for problem resolution by significant particpants in discord

We're essentially just a few Alpine nerds that work on these community projects when we're not working at our full time jobs. We don't get paid for this (feel free to sponsor us though!), and nothing we say should be taken as official AT ALL. 

I can see how that could be confusing though since we talk a lot on the Discord as well as the issues/discussions on the main repo. However, the buck stops with Caleb. Only what he says is canon.

For your CodePen frustrations, there's another project that adds "devtool" support to CodePen, which you can find [here](https://github.com/KevinBatdorf/alpine-inline-devtools) and a demo on CodePen [here](https://codepen.io/KevinBatdorf/pen/abdJGYM). I created that simply because I felt the same frustration you feel now when using CodePen + Alpine. 

Feel free to use that.

OR... you could figure out how to get it working in CodePen and submit a PR :) Everyone is welcome to contribute here.
OR... start building a better CodePen alternative for Alpine users and we can all work on that together.

**Reply by markfirmware** [reply]
> > a codepen is nearly universally requested for problem resolution by significant particpants in discord
> 
> We're essentially just a few Alpine nerds that work on these community projects when we're not working at our full time jobs. We don't get paid for this (feel free to sponsor us though!), and nothing we say should be taken as official AT ALL.
> 
> I can see how that could be confusing though since we talk a lot on the Discord as well as the issues/discussions on the main repo. However, the buck stops with Caleb. Only what he says is canon.
> 
> For your CodePen frustrations, there's another project that adds "devtool" support to CodePen, which you can find [here](https://github.com/KevinBatdorf/alpine-inline-devtools) and a demo on CodePen [here](https://codepen.io/KevinBatdorf/pen/abdJGYM). I created that simply because I felt the same frustration you feel now when using CodePen + Alpine.
> 
> Feel free to use that.
> 
> OR... you could figure out how to get it working in CodePen and submit a PR :) Everyone is welcome to contribute here.
> OR... start building a better CodePen alternative for Alpine users and we can all work on that together.

Ok, I understand about what is official (more accurately, what is not official) and your other points.

I am only using codepen because it was in my face in alpinejs land. Closire this week and give it a rest. I will study that codepen with inline dev tools.

FYI I have only one reason to use alpinejs and that is [It's what we use for the Tailwind UI site itself, and is without a doubt what I'd recommend if you'd otherwise be writing jQuery or vanilla JS.](https://tailwindui.com/documentation#js-alpine). I take tailwind as an internet axiom, for better or for worse.

Another FYI: I want a better gh readme. gh pages docs/index.html using just alpinejs and tailwind seems to be state of the art in simplicity.

> OR... you could figure out how to get it working in CodePen and submit a PR :) Everyone is welcome to contribute here.

I do not think that is the best fit for me.

> OR... start building a better CodePen alternative for Alpine users and we can all work on that together

I am interested. Maybe a fresh discussion for this?
